app/src/main/java/net/opendasharchive/openarchive/features/settings/ProofModeSettingsActivity.kt

```diff
@@ -67,8 +79,9 @@ class ProofModeSettingsActivity: BaseActivity() {
 
             val pkePreference = findPreference<SwitchPreferenceCompat>(Prefs.USE_PROOFMODE_KEY_ENCRYPTION)
             val activity = activity
+            val availability = Hbks.deviceAvailablity(requireContext())
```

Comment:
Just out of curiosity: Why do you use requireContext() (which makes the app crash) here instead of the activity from one line above and do the null check before?

Reply:
its the Kotlin way around default Java nullability (Context!); if you don't have a context here, you're probably pretty hooped anyway (Java code never checked)

I find it more clear than `context!!` and less obtuse than `context?.let {}`

A matter of opinion and standards though.

Reply:
Yeah, I know all these options. I typically try not to crash the app, that's why I asked.